File: backend/src/main/java/com/fitgym/model/Exercise.java
package com.fitgym.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.List;

@Document(collection = "exercises")
@Data
public class Exercise {
    @Id
    private String id;
    private String name;
    private String description;
    private String targetBodyPart;
    private String videoUrl;
    private String articleUrl;
    private String difficulty;
    private List<String> equipment;
}

File: backend/src/models/Exercise.js
const mongoose = require('mongoose');

const safeUrlValidator = {
  validator(value) {
    if (!value) return true;
    return /^https?:\/\/[^\s]+$/i.test(value);
  },
  message: 'URL fields must be valid http(s) URLs.'
};

const exerciseSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    description: String,
    targetBodyPart: { type: String, required: true, trim: true },
    videoUrl: { type: String, validate: safeUrlValidator },
    articleUrl: { type: String, validate: safeUrlValidator },
    difficulty: String,
    equipment: [String]
  },
  {
    collection: 'exercises',
    toJSON: {
      virtuals: true,
      transform: (_doc, ret) => {
        ret.id = ret._id.toString();
        delete ret._id;
        delete ret.__v;
        return ret;
      }
    }
  }
);

module.exports = mongoose.model('Exercise', exerciseSchema);
